drivers/at/at.c

```diff
@@ -334,132 +384,134 @@ ssize_t at_send_cmd_get_lines(at_dev_t *dev, const char *command,
 
     res = at_send_cmd(dev, command, timeout);
     if (res) {
-        goto out;
+        return res;
     }
 
     memset(resp_buf, '\0', len);
 
-    while (1) {
-        res = at_readline(dev, pos, bytes_left, keep_eol, timeout);
+    bool first_line = true;
+    while (bytes_left) {
+        if (first_line) {
+            res = at_readline_skip_empty(dev, pos, bytes_left, keep_eol, timeout);
+            first_line = false;
+        } else {
+            // keep subsequent empty lines, for whatever reason
+            res = at_readline(dev, pos, bytes_left, keep_eol, timeout);
+        }
         if (res == 0) {
-            if (bytes_left) {
-                *pos++ = eol[sizeof(eol) - 2];
-                bytes_left--;
-            }
+            *pos++ = eol[sizeof(eol) - 2];
+            bytes_left--;
             continue;
         }
         else if (res > 0) {
-            bytes_left -= res;
-            size_t len_ok = sizeof(CONFIG_AT_RECV_OK) - 1;
-            size_t len_error = sizeof(CONFIG_AT_RECV_ERROR) - 1;
-            if (((size_t )res == (len_ok + keep_eol)) &&
-                (len_ok != 0) &&
-                (strncmp(pos, CONFIG_AT_RECV_OK, len_ok) == 0)) {
+            size_t const res_len = res;
+            bytes_left -= res_len;
+            res = at_parse_resp(dev, pos);
+
+            switch (res) {
+            case 0: // OK
                 res = len - bytes_left;
-                break;
-            }
-            else if (((size_t )res == (len_error + keep_eol)) &&
-                     (len_error != 0) &&
-                     (strncmp(pos, CONFIG_AT_RECV_ERROR, len_error) == 0)) {
-                return -1;
-            }
-            else if (strncmp(pos, "+CME ERROR:", 11) == 0) {
-                return -2;
-            }
-            else if (strncmp(pos, "+CMS ERROR:", 11) == 0) {
-                return -2;
-            }
-            else {
-                pos += res;
-                if (bytes_left) {
-                    *pos++ = eol[sizeof(eol) - 2];
-                    bytes_left--;
-                }
-                else {
-                    return -1;
+                return res;
+            case 1: // response or URC
+                pos += res_len;
+                if (bytes_left == 0) {
+                    return -ENOBUFS;
                 }
+                *pos++ = eol[sizeof(eol) - 2];
+                bytes_left--;
+                continue;
+            default: // <0
+                return res;
             }
         }
         else {
-            break;
+            return res;
         }
     }
 
-out:
-    return res;
+    return -ENOBUFS;
 }
 
 int at_send_cmd_wait_prompt(at_dev_t *dev, const char *command, uint32_t timeout)
 {
-    unsigned cmdlen = strlen(command);
+    char *resp_buf = dev->err_buf;
+    size_t resp_buf_size = dev->err_buf_size;
```

Comment:
If you are using `err_buf` as a temporary buffer then you need to add a comment. Or, you need to rethink in what circumstances you need a buffer. The name suggests "error", which is not true in all cases where it is used.